### PR TITLE
Slideshow on front page of vivo

### PR DESCRIPTION
**Pull request by mrpatrickwest**
based on dco-viz repository. Still need to link to vivo pages for the larger visualizations instead of to the /dco-viz directory under apache root

### CODE REVIEW

**Comment by mrpatrickwest** [comment]
needs some styling changes for the pie charts because the titles don't completely display.

**Comment by zhongh** [comment]
@am-e Did you put the newest update on https://dco.tw.rpi.edu/vivo/? It just looks nice on there.

**Comment by zhongh** [comment]
I think it looks beautiful 👍

**Comment by olyerickson** [comment]
The slideshow at https://dco.tw.rpi.edu/vivo/ looks fantastic; great job!

**Comment by mrpatrickwest** [comment]
what is here is what we're seeing on https://dco.tw.rpi.edu/vivo/ so should be current.

**Comment by mrpatrickwest** [comment]
Are we okay with the new changes? Putting the js and css under VIVO instead of a separate repository? And should I remove the corresponding files from the dco-viz repository? Or keep them there for development purposes?

**Comment by zednis** [comment]
I don't see a reason to delete any files from the dco-viz repository, I just think that repository should be deprecated at some point.
- Are all CSS and JS resources for the slideshow now in the dco-vivo:slideshow branch?
- Why have themes/dco/js/subjectAreas.js as a separate file?  All it does is define a JS string variable
- Why do we have 2 different versions of jquery in /js?  jquery-1.7.2.min.js and jquery-2.1.4.js
- **perhaps put the slideshow FTL into its own FTL that is imported by the homepage FTL instead of embedded directly?**

**Comment by am-e** [comment]
- We do not need subjectAreas.js, It's a remnant of the older solution.
- As for the 2 different jquery versions, I had implemented the slideshow using jquery 2.1.4 but Hao's code for the statistics "requires" 1.7.2. I think we can just use 1.7.2, although I'd hoped we'd be able to use the latest.
- It would probably be a very smart move if we could create an FTL for the slideshow. 

Thanks Patrick for re-organizing the files into VIVO.

File: themes/dco/js/expertiseCountQuery.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (mrpatrickwest) [inline themes/dco/js/expertiseCountQuery.js]: do we need expertiseCountQuery.js? Seems same as subjectAreas.js, just a single variable defined.

**Comment by mrpatrickwest** [comment]
Can't we just link to an external jquery-1.7.2.min.js file instead of having our own? Or is it better to have our own?

File: themes/dco/js/expertiseCountQuery.js (hunk 1 above)
Comment (am-e) [inline themes/dco/js/expertiseCountQuery.js]: We do not need both. They're both how things were done before the static files.

**Comment by zednis** [comment]
@mrpatrickwest I think it makes sense to refer to a public CDN for jquery (any version).

**Comment by am-e** [comment]
In my experience https://cdnjs.cloudflare.com/ajax/libs/jquery/1.7.2/jquery.min.js is perfectly fine.

**Comment by mrpatrickwest** [comment]
@zednis @am-e I hesitate separating out the slideshow (and stats) from page-home.ftl because of how difficult it was to get everything in the right order and to be put in the right place. I would like to go ahead and push what we have and in the future we could separate it out when we have more time. We've already spent way too much time on this and need to get going on other tasks that are far more important.

I know we had to move the css for the slideshow above the javascript for some weird reason. And we would need to link to the js for the stats and the css for the slideshow in page-home.ftl.

I'm sure it probably would be "easy" to do ... but not right now. We can create another ticket to separate it out and if someone has any spare time to work on it they are more than welcome to.

**Comment by am-e** [comment]
Yes, I forgot the fact that getting the layout to fit and work well was hassle. Agreed.

**Comment by zednis** [comment]
Not addressing this now will just make it technical debt we will likely not address for a long time (if ever).

If you feel strongly about pushing this in the current state then create a backlog ticket to refactor and please put near the top of the queue.

My preference would be to not push this until after the refactor.

**Comment by mrpatrickwest** [comment]
that went a lot smoother than I thought it would.

Need to create a few tickets here in that there appears to be a way to set scripts and styles to be included in a list somewhere and have them loaded from that list instead of hardcoding it into the page-home.ftl.

**Comment by mrpatrickwest** [comment]
Can you both test again. Need to make sure this works in all the browsers

**Comment by am-e** [comment]
We're talking on dco.tw? I checked in Chrome, Firefox and Safari.
I caught one issue, in the alert when "more information" is clicked on the Members pie, the message contains ".. and undefined publications" I say since it is the members pie we take out the second part of the sentence. That is what I did in the publications pie.
Otherwise, looks good.

**Comment by zednis** [comment]
:+1: to the refactored slideshow CSS/JS

I tested the dco.tw main page slideshow with Chrome, Safari, and Firefox on OSX and it seemed to work in all.

There were some 404 errors in the web console:

> https://dco.tw.rpi.edu/dco-viz/css/result-light.css Failed to load resource: the server responded with a status of 404 (Not Found)
> jquery.min.js:4 GET https://dco.tw.rpi.edu/vivo/dco-viz/statsJSON/communityMemberCounts2.json 404 (Not Found)

**Comment by mrpatrickwest** [comment]
@zednis fixed the two problems you mentioned above. I get no errors or warnings in console now for any of the pages.

I say :shipit:

**Comment by zednis** [comment]
Looks good.  dco.tw.rpi.edu is running fast and I can confirm I see no errors.

:shipit:

**Comment by am-e** [comment]
All looks good even on the phone.

**Pull request merged**